File: Java/DownloadImagerrrr/app/src/main/java/com/example/downloadimage/MainActivity.java
package com.example.downloadimage;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.net.Uri;
import android.os.Bundle;
import android.provider.MediaStore;
import android.view.View;
import android.widget.Button;
import android.widget.ImageView;

import java.net.URI;

public class MainActivity extends AppCompatActivity {

    private final int GALLERY_REQ_CODE=1000;
    ImageView imgGallery,imgGallery2;
    Button btn_Download;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        imgGallery=findViewById(R.id.imgGallery);
        imgGallery2=findViewById(R.id.imgGallery2);
        btn_Download=findViewById(R.id.btn_Download);

        btn_Download.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {

                Intent iGallery=new Intent(Intent.ACTION_PICK);
                iGallery.setData(MediaStore.Images.Media.EXTERNAL_CONTENT_URI);
                startActivityForResult(iGallery,GALLERY_REQ_CODE);
            }
        });
    }

    @Override
    protected void onActivityResult(int requestCode, int resultCode, @Nullable Intent data) {
        super.onActivityResult(requestCode, resultCode, data);
        Uri uri=null;
        if(resultCode==RESULT_OK)
        {

            if(requestCode==GALLERY_REQ_CODE)

            {  imgGallery.setImageURI(data.getData());
                  uri=(Uri)data.getData();  //save uri for further processing
            }
            }
        imgGallery2.setImageURI(uri);
    }
}

File: Java/recylerpr/app/src/main/java/com/example/recylerpr/MainActivity.kt
package com.example.recylerpr

import android.content.Intent
import android.os.Bundle
import android.provider.MediaStore
import android.view.View
import android.widget.Button
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.RecyclerView
import androidx.recyclerview.widget.StaggeredGridLayoutManager

class MainActivity : AppCompatActivity() {
    //create list of int photos
    var photos = ArrayList<items_container>()
    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
        super.onActivityResult(requestCode, resultCode, data)
        if (resultCode == RESULT_OK) {
            if (requestCode == 1000) {
                val recyclerView = findViewById<RecyclerView>(R.id.rv)
                recyclerView.layoutManager = StaggeredGridLayoutManager(2, StaggeredGridLayoutManager.VERTICAL)

                //get uri of image
                //add uri to list
                data!!.data?.let { items_container(it) }?.let { photos.add(it) }
                //notify adapter
                val adapter = RecylerItemAdapter(this, photos)
                recyclerView.adapter = adapter
            }
        }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val recyclerView = findViewById<RecyclerView>(R.id.rv)
        recyclerView.layoutManager = StaggeredGridLayoutManager(2, StaggeredGridLayoutManager.VERTICAL)
        val add = findViewById<Button>(R.id.btnAddImage)
        add.setOnClickListener { v: View? ->
            val iGallery = Intent(Intent.ACTION_PICK)
            iGallery.data = MediaStore.Images.Media.EXTERNAL_CONTENT_URI
            startActivityForResult(iGallery, 1000)
        }
        val adapter = RecylerItemAdapter(this, photos)
        recyclerView.adapter = adapter

        //create adapter
    }
}